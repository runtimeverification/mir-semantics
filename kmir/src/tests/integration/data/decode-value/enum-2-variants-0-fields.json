{
  "bytes": [
    16
  ],
  "types": [],
  "typeInfo": {
    "EnumType": {
      "name": "Foo",
      "adt_def": 1,
      "discriminants": [
        255,
        16
      ],
      "fields": [
        [],
        []
      ],
      "layout": {
        "fields": {
          "Arbitrary": {
            "offsets": [
              {
                "num_bits": 0
              }
            ]
          }
        },
        "variants": {
          "Multiple": {
            "tag": {
              "Initialized": {
                "value": {
                  "Int": {
                    "length": "I8",
                    "signed": false
                  }
                },
                "valid_range": {
                  "start": 0,
                  "end": 1
                }
              }
            },
            "tag_encoding": "Direct",
            "tag_field": 0,
            "variants": [
              {
                "fields": {
                  "Arbitrary": {
                    "offsets": []
                  }
                },
                "variants": {
                  "Single": {
                    "index": 0
                  }
                },
                "abi": {
                  "Aggregate": {
                    "sized": true
                  }
                },
                "abi_align": 1,
                "size": {
                  "num_bits": 8
                }
              },
              {
                "fields": {
                  "Arbitrary": {
                    "offsets": []
                  }
                },
                "variants": {
                  "Single": {
                    "index": 1
                  }
                },
                "abi": {
                  "Aggregate": {
                    "sized": true
                  }
                },
                "abi_align": 1,
                "size": {
                  "num_bits": 8
                }
              }
            ]
          }
        },
        "abi": {
          "Scalar": {
            "Initialized": {
              "value": {
                "Int": {
                  "length": "I8",
                  "signed": false
                }
              },
              "valid_range": {
                "start": 0,
                "end": 1
              }
            }
          }
        },
        "abi_align": 1,
        "size": {
          "num_bits": 8
        }
      }
    }
  }
}
