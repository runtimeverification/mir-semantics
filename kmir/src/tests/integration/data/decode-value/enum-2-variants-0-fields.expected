Aggregate ( variantIdx ( 1 ) , .List )
